pkgs/os-specific/darwin/whatsapp/default.nix

```
@@ -0,0 +1,45 @@
+{ lib
+, fetchzip
+, stdenvNoCC
+, makeWrapper
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "whatsapp";
+  version = "2.24.11.85";
+
+  src = fetchzip {
+    extension = "zip";
+    name = "WhatsApp.app";
+    url = "https://web.whatsapp.com/desktop/mac_native/release/?version=${version}&extension=zip&configuration=Release&branch=relbranch";
+    sha256 = "sha256-vbgym7AfRi6kzZVMPRA6lP+6pL8cXnR7yt9hWPnt3+E=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{Applications,bin}
+    cp -r ../WhatsApp.app $out/Applications
+
+    makeWrapper $out/Applications/WhatsApp.app/Contents/MacOS/WhatsApp $out/bin/WhatsApp
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Native desktop client for WhatsApp";
+    homepage = "https://www.whatsapp.com/";
+    mainProgram = "WhatsApp";
+    license = licenses.unfree;
+    platforms = platforms.darwin;
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    maintainers = with maintainers; [ iivusly ];
```

```suggestion
  meta = {
    description = "Native desktop client for WhatsApp";
    homepage = "https://www.whatsapp.com/";
    license = lib.licenses.unfree;
    maintainers = with lib.maintainers; [ iivusly ];
    platforms = lib.platforms.darwin;
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```

Is using `with lib;` on `meta` is discouraged

https://nixos.org/manual/nixpkgs/stable/#chap-meta